File: apps/frontend/mobile/src/hooks/pool/poolView.ts
import type {
  MobilePoolDetail,
  MobilePoolSummary,
  MobilePoolUpdated,
  MobileTeamPool,
} from '@/src/api/generated/mobile-gateway/models';
import type {
  EnrichedPoolDTO,
  PoolSummaryDTO,
  PoolUpdateResult,
  RankedPoolView,
} from '@/src/types/Pool';
import {
  toCatalogDivisionView,
  toFormatView,
  toGenderView,
  toRankingTeamView,
} from '@/src/hooks/catalog/catalogView';

export function toRankedPoolView(response: MobileTeamPool): RankedPoolView {
  return {
    id: response.id,
    leagueCode: response.leagueCode,
    leagueName: response.leagueName,
    shortName: response.shortName,
    gender: toGenderView(response.gender),
    ranking: response.ranking.map(toRankingTeamView),
    division: toCatalogDivisionView(response.division),
  };
}

export function toEnrichedPoolView(
  response: MobilePoolDetail,
): EnrichedPoolDTO {
  return {
    id: response.id,
    season: response.season,
    leagueCode: response.leagueCode,
    leagueName: response.leagueName,
    name: response.name,
    shortName: response.shortName,
    rawName: response.rawName,
    gender: toGenderView(response.gender),
    followersCount: response.followersCount,
    ranking: response.ranking.map(toRankingTeamView),
    division: toCatalogDivisionView(response.division),
  };
}

export function toPoolSummaryView(response: MobilePoolSummary): PoolSummaryDTO {
  return {
    id: response.id,
    name: response.name,
    season: response.season,
    gender: toGenderView(response.gender),
    format: toFormatView(response.format),
    division: response.division
      ? toCatalogDivisionView(response.division)
      : null,
    leagueCode: response.leagueCode,
    leagueName: response.leagueName,
  };
}

export function toPoolUpdateResult(
  response: MobilePoolUpdated,
): PoolUpdateResult {
  return {
    id: response.id,
    name: response.name,
    shortName: response.shortName,
  };
}

File: apps/backend/mobile-gateway/src/main/java/com/blockout/mobilegateway/pool/application/views/PoolView.java
package com.blockout.mobilegateway.pool.application.views;

import com.blockout.mobilegateway.config.application.views.DivisionView;
import com.blockout.mobilegateway.shared.application.models.Format;
import com.blockout.mobilegateway.shared.application.models.Gender;
import com.blockout.mobilegateway.team.application.views.TeamWithStatsView;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PoolView {

    private Long id;

    private String season;

    private String poolCode;

    private String leagueCode;

    private String leagueName;

    private String name;

    private String shortName;

    private String rawName;

    private Format format;

    private Gender gender;

    private Long followersCount;

    private List<TeamWithStatsView> ranking;

    private DivisionView division;
}
